File: 728-SelfDividingNumbers/728-SelfDividingNumbers.py
# Last updated: 7/30/2025, 12:23:36 PM
class Solution:
    def selfDividingNumbers(self, left: int, right: int) -> List[int]:
        nums = []
        for i in range(left,right+1):
            temp = i
            ans = True
            while(temp>0):
                last = temp%10
                if(last == 0 or i%last != 0 ):
                    ans = False
                    break
                temp //= 10
            if ans :
                nums.append(i)
        return nums

File: 728-SelfDividingNumbers/728-SelfDividingNumbers.java
// Last updated: 7/30/2025, 12:23:42 PM
class Solution {
    public List<Integer> selfDividingNumbers(int left, int right) {
        List<Integer> nums = new ArrayList<>();
        for(int i=left; i<=right; i++){
            boolean ans = true;
            int temp = i;
            while(temp>0){
                if(temp%10 == 0 || i%(temp%10)!=0){
                    ans = false;
                    break;
                }
                temp /= 10;
            }
            if(ans){
                nums.add(i);
            }
        }
        return nums;
    }
}